File: src/main/java/diff/patch/accessors/SetGetter.java
package diff.patch.accessors;

import diff.patch.Getter;

import java.util.Set;

/**
 *
 * @author sebasjm <smarchano@primary.com.ar>
 */
public class SetGetter<Element> implements Getter<Element> {
    final Set<Element> set;
    final Element element;

    public SetGetter(Set<Element> set, Element element) {
        this.set = set;
        this.element = element;
    }

    @Override
    public Element get() {
        return element != null && set.contains(element) ? element : null;
    }
    
}

File: src/main/java/diff/patch/accessors/SetGetter.kt
package diff.patch.accessors

import diff.patch.Getter

/**

 * @author sebasjm <smarchano></smarchano>@primary.com.ar>
 */
class SetGetter<Element: Any>(internal val set: Set<Element>, internal val element: Element?) : Getter<Element> {

    override fun get(): Element? {
        return if (element != null && set.contains(element)) element else null
    }

}
